From 5cc4ccf5ca1b9bfaaf084b0fb2c22fc49f5da59e Mon Sep 17 00:00:00 2001
From: Philip Balister <philip@balister.org>
Date: Tue, 24 Sep 2013 09:36:36 -0400
Subject: [PATCH 35/40] gnuradio-runtime : Fix uninitialized variable.

Addresses Coverity 1043304. If either deserialize_untagged fails, goto the
error handler. Original code probably worked, since if one call failed, the
other would.

Signed-off-by: Philip Balister <philip@balister.org>
---
 gnuradio-runtime/lib/pmt/pmt_serialize.cc |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gnuradio-runtime/lib/pmt/pmt_serialize.cc b/gnuradio-runtime/lib/pmt/pmt_serialize.cc
index 4036b8d..56e92a1 100644
--- a/gnuradio-runtime/lib/pmt/pmt_serialize.cc
+++ b/gnuradio-runtime/lib/pmt/pmt_serialize.cc
@@ -577,7 +577,7 @@ deserialize(std::streambuf &sb)
   case PST_COMPLEX:
     {
     double r,i;
-    if(!deserialize_untagged_f64(&r, sb) && !deserialize_untagged_f64(&i, sb))
+    if(!deserialize_untagged_f64(&r, sb) || !deserialize_untagged_f64(&i, sb))
       goto error;
     return make_rectangular( r,i );
     }
-- 
1.7.10.4

